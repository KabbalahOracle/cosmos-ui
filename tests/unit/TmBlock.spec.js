import { mount } from "@vue/test-utils"
import htmlBeautify from "html-beautify"
import { TmBlock } from "@/index"

describe("TmBlock.vue", () => {
  let wrapper
  beforeEach(async () => {
    wrapper = mount(TmBlock, {
      propsData: {}
    })
  })

  it("has the expected html structure", () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })
})
