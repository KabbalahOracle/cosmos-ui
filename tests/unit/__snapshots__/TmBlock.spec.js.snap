// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmBlock.vue has the expected html structure 1`] = `
<div>
  <section class="tm-part">
    <div class="tm-part-container">
      <!---->
      <main class="tm-part-main">
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Block Hash
              </div>
              <div class="tm-li-dd">
                asdf
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
  <section class="tm-part">
    <div class="tm-part-container">
      <header class="tm-part-header">
        <div class="tm-part-header-container">
          <div class="tm-part-title h5">
            Header
          </div>
          <menu class="tm-part-menu"></menu>
        </div>
      </header>
      <main class="tm-part-main">
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Chain ID
              </div>
              <div class="tm-li-dd">
                1
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Time
              </div>
              <div class="tm-li-dd">
                January 5th, 1988
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Transactions
              </div>
              <div class="tm-li-dd">
                3
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Last Commit Hash
              </div>
              <div class="tm-li-dd">
                4
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Validators Hash
              </div>
              <div class="tm-li-dd">
                5
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                App Hash
              </div>
              <div class="tm-li-dd">
                6
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
  <section class="tm-part">
    <div class="tm-part-container">
      <header class="tm-part-header">
        <div class="tm-part-header-container">
          <div class="tm-part-title h5">
            Last Block
          </div>
          <menu class="tm-part-menu"></menu>
        </div>
      </header>
      <main class="tm-part-main">
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Hash
              </div>
              <div class="tm-li-dd">
                7
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Parts Total
              </div>
              <div class="tm-li-dd">
                8
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Parts Hash
              </div>
              <div class="tm-li-dd">
                9
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
  <section class="tm-part">
    <div class="tm-part-container">
      <header class="tm-part-header">
        <div class="tm-part-header-container">
          <div class="tm-part-title h5">
            Precommit
          </div>
          <menu class="tm-part-menu"></menu>
        </div>
      </header>
      <main class="tm-part-main">
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Address
              </div>
              <div class="tm-li-dd">
                10
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Index
              </div>
              <div class="tm-li-dd">
                11
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Round
              </div>
              <div class="tm-li-dd">
                12
              </div>
            </div>
          </div>
        </div>
        <div class="tm-li" name="proposal">
          <div class="tm-li-container">
            <div class="tm-li-thumb">
              <!---->
            </div>
            <div class="tm-li-dl">
              <div class="tm-li-dt">
                Sig (13)
              </div>
              <div class="tm-li-dd">
                14
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
  <section class="tm-part">
    <div class="tm-part-container">
      <header class="tm-part-header">
        <div class="tm-part-header-container">
          <div class="tm-part-title h5">
            Transactions
          </div>
          <menu class="tm-part-menu"></menu>
        </div>
      </header>
      <main class="tm-part-main">
        <div class="tm-li-tx">
          <div class="tm-li-tx__icon">
            <img src="~assets/cosmos-logo.png" style="border-color: #25293f;">
          </div>
          <div class="tm-li-tx__content">
            <h1 class="tm-li-tx__content__caption">
              <div>
                Sent&nbsp;
                <b>1234</b><span>&nbsp;JBCOINS</span>
                <!---->
              </div>
            </h1>
            <div class="tm-li-tx__content__information">
              <div class="tm-li-tx__content__information__time">
                1:00
              </div>
              <div class="tm-li-tx__content__information__details">
                <span>To tb1d4u5zerywfjhxuc9nudvw</span>
              </div>
              <a href="https://explorecosmos.network/blocks/3436" class="tm-li-tx__content__information__block">Block 3436</a>
            </div>
          </div>
        </div>
        <div class="tm-li-tx">
          <div class="tm-li-tx__icon">
            <img src="~assets/cosmos-logo.png" style="border-color: #215c2f;">
          </div>
          <div class="tm-li-tx__content">
            <h1 class="tm-li-tx__content__caption">
              <div>
                Received&nbsp;
                <b>1234</b><span>&nbsp;JBCOINS</span>
              </div>
            </h1>
            <div class="tm-li-tx__content__information">
              <div class="tm-li-tx__content__information__time">
                1:00
              </div>
              <div class="tm-li-tx__content__information__details">
                <span>From tb1d4u5zerywfjhxuc9nudvw</span>
              </div>
              <a href="https://explorecosmos.network/blocks/3438" class="tm-li-tx__content__information__block">Block 3438</a>
            </div>
          </div>
        </div>
        <div class="tm-li-tx">
          <div class="tm-li-tx__icon">
            <img src="~assets/cosmos-logo.png" style="border-color: #25293f;">
          </div>
          <div class="tm-li-tx__content">
            <h1 class="tm-li-tx__content__caption">
              <div>
                Sent&nbsp;
                <b>1234</b><span>&nbsp;JBCOINS</span>&nbsp;(to yourself)
              </div>
            </h1>
            <div class="tm-li-tx__content__information">
              <div class="tm-li-tx__content__information__time">
                1:00
              </div>
              <div class="tm-li-tx__content__information__details">
                <span>To tb1da6xsetjg9jxgun9wdesexv05j</span>
              </div>
              <a href="https://explorecosmos.network/blocks/466" class="tm-li-tx__content__information__block">Block 466</a>
            </div>
          </div>
        </div>
        <div class="tm-li-tx">
          <div class="tm-li-tx__icon">
            <img src="~assets/cosmos-logo.png" style="border-color: #512584;">
          </div>
          <div class="tm-li-tx__content">
            <h1 class="tm-li-tx__content__caption">
              <div>
                Delegated&nbsp;
                <b>4200</b><span>&nbsp;STEAK</span>
              </div>
            </h1>
            <div class="tm-li-tx__content__information">
              <div class="tm-li-tx__content__information__time">
                1:00
              </div>
              <div class="tm-li-tx__content__information__details">
                <div>
                  To&nbsp;
                  <a href="/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw">cool validator</a>
                </div>
              </div>
              <a href="https://explorecosmos.network/blocks/566" class="tm-li-tx__content__information__block">Block 566</a>
            </div>
          </div>
        </div>
        <div class="tm-li-tx">
          <div class="tm-li-tx__icon">
            <img src="~assets/cosmos-logo.png">
          </div>
          <div class="tm-li-tx__content">
            <h1 class="tm-li-tx__content__caption">
              <span>Unknown Transaction Type</span>
            </h1>
            <div class="tm-li-tx__content__information">
              <div class="tm-li-tx__content__information__time">
                1:00
              </div>
              <div class="tm-li-tx__content__information__details"></div>
              <a href="https://explorecosmos.network/blocks/566" class="tm-li-tx__content__information__block">Block 566</a>
            </div>
          </div>
        </div>
      </main>
    </div>
  </section>
</div>
`;
